### PR TITLE
[FLINK-22844]Add doc to introduce ExplainDetails for EXPLAIN sytnax

### PR DESCRIPTION
**Pull request by chaozwn**
## What is the purpose of the change

*(For example: This pull request makes task deployment go through the blob server, rather than through RPC. That way we avoid re-transferring them on each deployment (during recovery).)*

## Brief change log

*(for example:)*
  - *The TaskInfo is stored in the blob store on job creation time as a persistent artifact*
  - *Deployments RPC transmits only the blob storage reference*
  - *TaskManagers retrieve the TaskInfo from the blob cache*

## Verifying this change

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (100MB)*
  - *Extended integration test for recovery after master (JobManager) failure*
  - *Added test that validates that TaskInfo is transferred only once across recoveries*
  - *Manually verified the change by running a 4 node cluser with 2 JobManagers and 4 TaskManagers, a stateful streaming program, and killing one JobManager and two TaskManagers during the execution, verifying that recovery happens correctly.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no)
  - The serializers: (yes / no / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / no / don't know)
  - The S3 file system connector: (yes / no / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: docs/content.zh/docs/dev/table/sql/explain.md [hunk 1]
```diff
@@ -30,6 +30,7 @@ under the License.
 
 EXPLAIN statements are used to explain the logical and optimized query plans of a query or an INSERT statement.
 
+
```
Comment (leonardBang) [inline docs/content.zh/docs/dev/table/sql/explain.md:33]: revert the unnecessary  change

File: docs/content.zh/docs/dev/table/sql/explain.md [hunk 2]
```diff
@@ -138,52 +153,144 @@ table_result = t_env.execute_sql(
...
+```
+
+{{< /tab >}}
+
+{{< /tabs >}}
+
 {{< top >}}
 
+## ExplainDetails
+```sql
```
Comment (leonardBang) [inline docs/content.zh/docs/dev/table/sql/explain.md:274]: The quoted content is not SQL

File: docs/content.zh/docs/dev/table/sql/explain.md [hunk 3]
```diff
@@ -138,52 +153,144 @@ table_result = t_env.execute_sql(
...
+ESTIMATED_COST: generates cost information on physical node estimated by optimizer, 
+e.g. TableSourceScan(..., cumulative cost ={1.0E8 rows, 1.0E8 cpu, 2.4E9 io, 0.0 network, 0.0 memory})
+
+CHANGELOG_MODE:generates changelog mode for every physical rel node. 
+e.g. GroupAggregate(..., changelogMode=[I,UA,D])
+
+JSON_EXECUTION_PLAN: generates the execution plan in json format of the program.
+```
+
+
```
Comment (leonardBang) [inline docs/content.zh/docs/dev/table/sql/explain.md:286]: useless

File: docs/content.zh/docs/dev/table/sql/explain.md [hunk 4]
```diff
@@ -138,52 +153,144 @@ table_result = t_env.execute_sql(
...
 ## Syntax
 
 ```sql
-EXPLAIN PLAN FOR <query_statement_or_insert_statement>
+EXPLAIN [([ExplainDetail[, ExplainDetail]*]) | PLAN FOR] <query_statement_or_insert_statement>
 ```
 
+
```
Comment (leonardBang) [inline docs/content.zh/docs/dev/table/sql/explain.md:294]: useless

File: docs/content/docs/dev/table/sql/explain.md [hunk 5]
```diff
@@ -139,52 +153,144 @@ table_result = t_env.execute_sql(
...
 ```
 
+{{< /tab >}}
+
+{{< /tabs >}}
+
 {{< top >}}
 
+## ExplainDetails
+```text
```
Comment (leonardBang) [inline docs/content/docs/dev/table/sql/explain.md:274]: as above

File: docs/content/docs/dev/table/sql/explain.md [hunk 6]
```diff
@@ -139,52 +153,144 @@ table_result = t_env.execute_sql(
...
+ESTIMATED_COST: generates cost information on physical node estimated by optimizer, 
+e.g. TableSourceScan(..., cumulative cost ={1.0E8 rows, 1.0E8 cpu, 2.4E9 io, 0.0 network, 0.0 memory})
+
+CHANGELOG_MODE:generates changelog mode for every physical rel node. 
+e.g. GroupAggregate(..., changelogMode=[I,UA,D])
+
+JSON_EXECUTION_PLAN: generates the execution plan in json format of the program.
+```
+
+
```
Comment (leonardBang) [inline docs/content/docs/dev/table/sql/explain.md:286]: as above

**Review by leonardBang — COMMENTED** [review]
Thanks @chaozwn for the contribution, the content looks good, I left some syntax comments.

**Comment by leonardBang** [comment]
@chaozwn Thanks for the update, but when we resolve reviewer's comment, we should append commit rather than squashing and force pushing which we cannot know the comment is fixed or not, because the squash way can not compare with previous comment  and reviewer have to review again.

**Comment by chaozwn** [comment]
> @chaozwn Thanks for the update, but when we resolve reviewer's comment, we should append commit rather than squashing and force pushing which we cannot know the comment is fixed or not, because the squash way can not compare with previous comment and reviewer have to review again.

I'm very sorry about do this.The reason why I use force push is because the latest version of flink is too big, and I need to re-merge a large number of branches in the code, and this may cause the display of this branch to be abnormal.

**Comment by chaozwn** [comment]
> @chaozwn Thanks for the update, but when we resolve reviewer's comment, we should append commit rather than squashing and force pushing which we cannot know the comment is fixed or not, because the squash way can not compare with previous comment and reviewer have to review again.

in next time, I will choose rebase master carefully

**Comment by leonardBang** [comment]
Yes, rebasing master is recommended, and it doesn't matter.

File: docs/content.zh/docs/dev/table/sql/explain.md [hunk 7]
```diff
@@ -137,52 +152,142 @@ table_result = t_env.execute_sql(
...
 {{< /tab >}}
 {{< tab "SQL CLI" >}}
+
 ```sql
-Flink SQL> CREATE TABLE MyTable1 (`count` bigint, word VARCHAR(256);
+Flink SQL> CREATE TABLE MyTable1 (`count` bigint, word VARCHAR(256)) WITH ('connector' = 'datagen');
 [INFO] Table has been created.
 
-Flink SQL> CREATE TABLE MyTable2 (`count` bigint, word VARCHAR(256);
+Flink SQL> CREATE TABLE MyTable2 (`count` bigint, word VARCHAR(256)) WITH ('connector' = 'datagen');
```
Comment (leonardBang) [inline docs/content.zh/docs/dev/table/sql/explain.md:170]: Nice change, could you also add connector option for other languages?

File: docs/content.zh/docs/dev/table/sql/explain.md [hunk 8]
```diff
@@ -137,52 +152,142 @@ table_result = t_env.execute_sql(
...
+> SELECT `count`, word FROM MyTable2;
 ```
 {{< /tab >}}
 {{< /tabs >}}
 
 The `EXPLAIN` result is:
+
+{{< tabs "explain result" >}}
+
+{{< tab "PLAN FOR" >}}
```
Comment (leonardBang) [inline docs/content.zh/docs/dev/table/sql/explain.md:189]:
```suggestion
{{< tab "EXPLAIN PLAN" >}}
```

File: docs/content.zh/docs/dev/table/sql/explain.md [hunk 9]
```diff
@@ -137,52 +152,142 @@ table_result = t_env.execute_sql(
...
+== Optimized Execution Plan ==
+Union(all=[true], union=[count, word])
+:- Calc(select=[count, word], where=[LIKE(word, _UTF-16LE'F%')])
+:  +- TableSourceScan(table=[[default_catalog, default_database, MyTable1]], fields=[count, word])
++- TableSourceScan(table=[[default_catalog, default_database, MyTable2]], fields=[count, word])
+```
+
+{{< /tab >}}
+
+{{< tab "EXPLAINDETAILS" >}}
```
Comment (leonardBang) [inline docs/content.zh/docs/dev/table/sql/explain.md:215]:
```suggestion
{{< tab "EXPLAIN PLAN WITH DETAILS" >}}
```

**Review by leonardBang — COMMENTED** [review]
@chaozwn Sorry for the late response,  I left minor comments, Could you also update your commit message which should be `[FLINK-22844]` ?

File: docs/content.zh/docs/dev/table/sql/explain.md (hunk 9 above)
Comment (chaozwn) [reply docs/content.zh/docs/dev/table/sql/explain.md]: I have completed the modification.

**Review by leonardBang — APPROVED** [review]
Thanks for he update @chaozwn , LGTM

**Comment by wuchong** [comment]
Merging...

**Pull request merged**